core/java/iesi-core/src/main/java/io/metadew/iesi/script/action/http/HttpExecuteRequest.java

```diff
@@ -132,6 +148,102 @@ protected boolean executeAction() throws NoSuchAlgorithmException, IOException,
+            stream = getParams(dataType.toString())
+                    .map(text -> getHeader(buildParam(text)));
+        } else if (dataType instanceof InMemoryDatasetImplementation) {
+            stream = getDataItems(dataType).entrySet().stream()
+                    .map(dataItem -> getHeader(buildParam(dataItem.getKey(), dataItem.getValue().toString())));
+        }
+        else {
+            log.warn(MessageFormat.format(getActionExecution().getAction().getType().concat(" does not accept {0} as type for expectedStatusCode"),
+                    dataType.getClass()));
+            return new ArrayList<>();
+        }
+
+        return stream.filter(Optional::isPresent).map(Optional::get).collect(Collectors.toList());
+    }
+
+    private List<HttpQueryParameter> convertQueryParams(DataType dataType) {
+        Stream<Optional<HttpQueryParameter>> stream;
+        if (dataType == null) {
+            return new ArrayList<>();
+        } else if (dataType instanceof Text){
+            stream = getParams(dataType.toString())
+                    .map(text -> getQueryparam(buildParam(text)));
+        } else if (dataType instanceof InMemoryDatasetImplementation) {
+            stream = getDataItems(dataType).entrySet().stream()
+                    .map(dataItem -> getQueryparam(buildParam(dataItem.getKey(), dataItem.getValue().toString())));
+        }
+        else {
+            log.warn(MessageFormat.format(getActionExecution().getAction().getType().concat(" does not accept {0} as type for expectedStatusCode"),
+                    dataType.getClass()));
+            return new ArrayList<>();
+        }
+        return stream.filter(Optional::isPresent).map(Optional::get).collect(Collectors.toList());
+    }
+
+    private Stream<String> getParams(String paramsStr) {
+        List<String> params = new ArrayList<>(Arrays.asList(paramsStr.split(",")));
+        return params.stream();
+    }
+
+    private Map<String, DataType> getDataItems(DataType dataType) {
+        return InMemoryDatasetImplementationService
+                .getInstance()
+                .getDataItems((InMemoryDatasetImplementation) dataType, getExecutionControl()
+                        .getExecutionRuntime());
+    }
+
+
+    private Optional<HttpHeader> getHeader(AbstractMap.SimpleEntry<String, String> paramEntry) {
+        if (paramEntry != null) {
+            return Optional.of(new HttpHeader(paramEntry.getKey(), paramEntry.getValue()));
+        }
+        return Optional.empty();
+    }
+
+    private Optional<HttpQueryParameter> getQueryparam(AbstractMap.SimpleEntry<String, String> paramEntry) {
+        if (paramEntry != null) {
+            return Optional.of(new HttpQueryParameter(paramEntry.getKey(), paramEntry.getValue()));
+        }
+        return Optional.empty();
+    }
+
+
+    private AbstractMap.SimpleEntry<String, String> buildParam(String param) {
+        List<String> keyValues;
+        String key;
+        String value;
```

Comment:
don't create these variables, they are just assigned once and immediatly used afterwards